packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1414,6 +1425,116 @@ export class MergeTree {
         return { segment, offset };
     }
 
+    private getSlideToSegment(currentSegment: ISegment) {
+        // Slide to the next farthest valid segment in the tree. If no such segment is found
+        // slide to the last valid segment.
+        // TODO this walks the whole tree to find the segment - could write a more efficient
+        // walk that starts at the segment
+        let foundStart = false;
+        let foundSegmentPastStart = false;
+        let slideToSegment: ISegment | undefined;
+        this.walkAllSegments(this.root, (seg) => {
+            if (seg.seq !== UnassignedSequenceNumber && !isRemovedAndAcked(seg)) {
+                slideToSegment = seg;
+                if (foundStart) {
+                    foundSegmentPastStart = true;
+                    return false;
+                }
+            }
+            if (!foundStart && seg === currentSegment) {
+                foundStart = true;
+            }
+            return true;
+        });
+        let offset = 0;
+        if (slideToSegment && !foundSegmentPastStart && !isRemoved(slideToSegment)) {
+            // If slid nearer onto a non-removed segment, offset should be at the end of the segment
+            offset = slideToSegment.cachedLength - 1;
+        }
+        return { segment: slideToSegment, offset };
+    }
+
+    /**
+     * @internal - this method should only be called by client
+     */
+    public getSlideOnRemoveReferenceSegmentAndOffset(pos: number, refSeq: number, clientId: number) {
+        let segoff = this.getContainingSegment(pos, refSeq, clientId);
+        if (segoff.segment && isRemovedAndAcked(segoff.segment)) {
+            // Only slide is the segment is removed and acked
```

Comment:
```suggestion
            // Only slide if the segment is removed and acked
```

(omitting the comment seems fine too... intent is clear with the helper you use in this block)